lib/vagrant-service-manager/config.rb

```diff
@@ -0,0 +1,39 @@
+require 'set'
+
+module Vagrant
+  module ServiceManager
+    class Config < Vagrant.plugin('2', :config)
+      attr_accessor :providers
+
+      PROVIDERS = ['docker']
+
+      def initialize
+        super
+        @providers = UNSET_VALUE
+      end
+
+      def finalize!
+        # enable docker as provider by default
+        @providers = 'docker' if @providers == UNSET_VALUE
+      end
+
+      def validate(machine)
+        errors = _detected_errors
+        errors.concat(validate_providers)
+        { "servicemanager" => errors }
+      end
+
+      private
+
+      def validate_providers
+        errors = []
+
+        if !Set[@providers.split(',').map(&:strip)].subset?(Set[PROVIDERS])
```

Comment:
if specified provider in configuration is not listed in `PROVIDERS` then it errors out.